src/Lib/App.hs

```diff
@@ -40,8 +40,5 @@ runAppAsHandler env action = do
                $ unApp action
 
   case res of
-    Left (Invalid text)     -> throwError $ err400 { errBody = textToLBS text }
-    Left NotFound           -> throwError err404
-    Left (NotAllowed text)  -> throwError $ err401 { errBody = textToLBS text }
-    Left (ServerError text) -> throwError $ err500 { errBody = textToLBS text }
-    Right a                 -> return a
+    Left appError -> throwError $ toHttpError appError
+    Right a       -> return a
```

Comment:
This `case` can be written simpler using `either` function:

* https://hackage.haskell.org/package/base-4.11.1.0/docs/Prelude.html#v:either

Also, it's better to use `pure` instead of `return`